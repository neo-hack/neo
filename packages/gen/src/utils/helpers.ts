import { Workflow } from '../interface'
import { hooks } from '../utils/hooks'

import Observable from 'zen-observable'
import type Listr from 'listr'
import { LIFE_CYCLES } from 'src/constants'

/**
 * @description convert schema to listr
 */
export const toListr = (schema: Workflow) => {
  const tasks: Listr.ListrTask[] = Object.keys(schema.jobs || {}).map((key) => {
    const taskName = schema.jobs![key].name || key
    return {
      title: taskName,
      task: () => {
        let ob: ZenObservable.SubscriptionObserver<any>
        const OB = new Observable<any>((observer) => {
          ob = observer
        })
        hooks.addHooks({
          [LIFE_CYCLES.STEP]: (args: { step: string }) => {
            ob.next(args.step)
          },
          [taskName]: () => {
            ob.complete()
          },
        })
        return OB as any
      },
    }
  })
  return tasks
}
