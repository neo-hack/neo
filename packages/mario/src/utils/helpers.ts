import { Workflow } from '../interface'
import { hooks } from './hooks'

import Observable from 'zen-observable'
import type Listr from 'listr'
import { LIFE_CYCLES } from 'src/constants'

/**
 * @description convert schema to listr
 */
export const toListr = (schema: Workflow) => {
  const tasks: Listr.ListrTask[] = Object.keys(schema.jobs || {}).map((key) => {
    const taskName = schema.jobs![key].name || key
    return {
      title: taskName,
      task: () => {
        let ob: ZenObservable.SubscriptionObserver<any>
        const OB = new Observable<any>((observer) => {
          ob = observer
        })
        hooks.addHooks({
          [LIFE_CYCLES.STEP]: (args: { step: string }) => {
            ob.next(args.step)
          },
          [taskName]: (args: { error: Error }) => {
            if (!args.error) {
              ob.complete()
              return
            }
            ob.error(args.error)
          },
        })
        return OB as any
      },
    }
  })
  return tasks
}
