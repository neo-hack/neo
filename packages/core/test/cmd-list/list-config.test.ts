import { execNeo, mockLockFile, storeDir } from '../helpers'

beforeAll(async () => {
  await mockLockFile()
})

describe('command list config', () => {
  it('list configs should list all', async () => {
    const { stdout } = await execNeo([
      'list',
      'configs',
      '--store-dir',
      storeDir,
      '--no-interactive',
      true,
    ])
    expect(stdout).toMatchSnapshot()
  })

  it('list configs should list filtered by preset', async () => {
    const { stdout } = await execNeo([
      'list',
      'configs',
      '--store-dir',
      storeDir,
      '--no-interactive',
      true,
      '--preset',
      'neo',
    ])
    expect(stdout).toMatchSnapshot()
  })
})
