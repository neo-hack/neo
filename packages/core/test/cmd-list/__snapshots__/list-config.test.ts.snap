// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`command list config list configs should list all 1`] = `
"Found 1 configs:

neo
CI"
`;

exports[`command list config list configs should list filtered by preset 1`] = `
"Found 1 configs:

CI"
`;
